packages/xo-server/src/xo-mixins/backups-ng-logs.js

```diff
@@ -65,139 +71,197 @@ const taskTimeComparator = ({ start: s1, end: e1 }, { start: s2, end: e2 }) => {
 //   status: 'pending' | 'failure' | 'interrupted' | 'skipped' | 'success',
 //   tasks?: Task[],
 // }
-export default {
-  getBackupNgLogs: debounceWithKey(
-    async function getBackupNgLogs(runId?: string) {
-      const [jobLogs, restoreLogs, restoreMetadataLogs] = await Promise.all([
-        this.getLogs('jobs'),
-        this.getLogs('restore'),
-        this.getLogs('metadataRestore'),
-      ])
-
-      const { runningJobs, runningRestores, runningMetadataRestores } = this
-      const consolidated = {}
-      const started = {}
-
-      const handleLog = ({ data, time, message }, id) => {
-        const { event } = data
-        if (event === 'job.start') {
-          if (
-            (data.type === 'backup' || data.key === undefined) &&
-            (runId === undefined || runId === id)
-          ) {
-            const { scheduleId, jobId } = data
-            consolidated[id] = started[id] = {
-              data: data.data,
-              id,
-              jobId,
-              jobName: data.jobName,
-              message: 'backup',
-              scheduleId,
-              start: time,
-              status: runningJobs[jobId] === id ? 'pending' : 'interrupted',
-            }
-          }
-        } else if (event === 'job.end') {
-          const { runJobId } = data
-          const log = started[runJobId]
-          if (log !== undefined) {
-            delete started[runJobId]
-            log.end = time
-            log.status = computeStatusAndSortTasks(
-              getStatus((log.result = data.error)),
-              log.tasks
-            )
-          }
-        } else if (event === 'task.start') {
-          const task = {
+export default class BackupNgLogs {
+  constructor(app, { backup }) {
+    this._app = app
+    app.on('clean', () =>
+      app.getStore(STORE_NAMESPACE).then(db => db.gc(backup.nKeptTasks))
+    )
+  }
+
+  @debounceWithKey.decorate(10e3, function keyFn(runId) {
+    return [this, runId]
+  })
+  async getBackupNgLogs(runId?: string) {
+    const app = this._app
+    const backupTaskStore = await app.getStore(STORE_NAMESPACE)
+
+    const consolidated = await backupTaskStore.getAll()
+    // an optimization to avoid useless computation
```

Comment:
Unclear and tautological, this would be better:
```suggestion
    // if the requested log is already consolidated, simply returns it
```